sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -356,6 +361,10 @@ public R block(Duration timeout) {
             throw logger.logExceptionAsWarning(new IllegalArgumentException(
                 "Negative or zero value for timeout is not allowed."));
         }
+        if (matchStatus(pollResponse, statusToBlockFor) || isComplete()){
```

Comment:
Is there are defined progression for status transition? For e.g. if the progression is (not_started -> started -> in_progress -> completed) and the current poll status is `in_progress` and the user called `blockUntil(started)` what is the expected result here? The status has gone past the `started` stage and is `in_progress` now, so should we just return or wait for timeout?

Reply:
Progression happens from Not Started -> In progress -> (Completed/Cancelled/Failed).
Users also can have their custom values inserted.
The only check we can add is if user says blockUntil(Not_Started), but that's redundant as that is the start state by default.

Reply:
So, what happens if I call `blockUntil(In_Progress)` if the status is already `Cancelled` or `Completed`?

Reply:
We kind of have a sequence which is not defined in any structure as of today.
NOT_STARTED 2 IN_PROGRESS 3. SUCCESSFULLY_COMPLETED/FAILED/USER_CANCELLED
The current behavior will be "it will timeout" in this case.
If we want to change this, A unanswered question will be which pollResponse we return, since the status is already gone past and we do not have that old pollResponse.

Reply:
if you call blockUntil(In_Progress) and poller is already completed/cancelled/failed then
it will just return that final last response.
The Javadoc mentions this.

Reply:
This is basic design question, we have not handelled, blockUntil() only make sense for a event which will happen in future.
The those event which happened in past : we could throw exception `StateAlreadyReachedException` . I feel like we do want more time to go though design discussion on this, may be take it later after GA.